lib/link_checker.rb

```diff
@@ -0,0 +1,46 @@
+require 'yaml'
+require 'rest-client'
+require_relative 'root'
+require_relative 'app_config'
+require_relative 'url_checker'
+
+
+
+class LinkChecker
+  def initialize
+    @urls = YAML.load_file(GOVUK_PAGES_FILE)
```

Comment:
I personally think `attr_reader` `attr_accessor`, make things more readable. And this seems to be the untold standard. You can make them private by placing them under the `private` keyword.